src/Quests/CustomQuest.cs

```diff
@@ -33,8 +33,13 @@ public class CustomQuest
         public bool Cancelable { get; set; }
         public string ReactionText { get; set; }
         public int DaysLeft { get; set; } = 0;
-        public List<Hook> Hooks { get; set; }
-
+        public List<Hook> Hooks { get; set; }
+        public string AddConversationTopicWhenQuestAccepted { get; set; }

```

Comment:
I think it's useless to define these fields on `CustomQuest`. Custom quest has an event `Completed`. I suggest add more events on `CustomQuest` like `Accepted` and `Removed` and call from QuestLogWatchdog for managed quests. The `Completed` event already works.

When you have all these needed events, then hook up listeners on them in `ContentPacks.Loader` class in the part of code where custom quests are created from content pack's json data and to these listeners call coresponding methods from your `ConversationTopic` class. It will be more clean.

This changes makes it works from ContentPacks. For programatic usage is not needed hook up listeners with add conversation topic methods, because SMAPI modder which uses C# for modding can hook up their won losteneres to `CustomQuest`'s events and call methods for work with topics on `Game1.player.activeDialogueEvents` manually.

Reply:
Quite lost with this one 😅
I tried to imitate the Complete method in CustomQuest for the Accepted and Removed. And after that I'm clueless...